### PR TITLE
Fork tracking

### PR DESCRIPTION
**Pull request by judepereira**

### CODE REVIEW

File: cli/cli.py [hunk 1]
```diff
@@ -41,23 +41,60 @@ def get_access_token():
...
+    if not enable_reporting:
+        return
+
+    try:
+        requests.post(url, json=json_data)
     except Exception:
         # ignore issues with tracking
         pass
 
+    if pid == 0:
```
Comment (wilzbach) [inline cli/cli.py]: This check is not needed and would only be partially covered as pid is always `0` at this point.

File: cli/cli.py [hunk 2]
```diff
@@ -127,36 +164,39 @@ def initiate_login():
...
             except IOError:
                 time.sleep(0.5)
                 # just try again
-                pass
+                continue
             except KeyboardInterrupt:
                 click.echo('Login failed. Please try again.')
                 sys.exit(1)
+
+        if res.json().get('beta') is False:
```
Comment (wilzbach) [inline cli/cli.py]: Shouldn't this be outside of the loop?

File: cli/cli.py (hunk 1 above)
Comment (judepereira) [reply cli/cli.py]: Yea, I already figured that. Didn't push.

File: cli/cli.py (hunk 2 above)
Comment (judepereira) [reply cli/cli.py]: No, notice the break below.

Comment (wilzbach) [reply cli/cli.py]: I noticed, but that's the only break. So you could just have the `break` here and the text outside of the loop ;-)

File: cli/cli.py [hunk 3]
```diff
@@ -41,23 +41,67 @@ def get_access_token():
...
+    forking the process.
 
-        extra['CLI version'] = version
+    The child makes the call, and does not return back to the caller,
+    but instead just exits (sys.exit(0)). This is to ensure that whatever
+    needs to happen after this method is called, does not happen twice.
+    """
+    if not enable_reporting:
+        return
+    if sys.platform != 'linux' and sys.platform != 'darwin':
```
Comment (wilzbach) [inline cli/cli.py]: There's also FreeBSD, DragonFlyBSD, NetBSD, OpenBSD and a lot of other Unix-like platform ;-)

Comment (judepereira) [reply cli/cli.py]: That's okay.

**Pull request merged**